Ical.Net/CalendarComponents/CalendarEvent.cs

```diff
@@ -42,132 +54,136 @@ public class CalendarEvent : RecurringComponent, IAlarmContainer, IComparable<Ca
-    /// Likewise, if the end time and duration is
-    /// available, but a start time is not determined,
-    /// the start time will be extrapolated from
-    /// available information.
-    /// </note>
-    /// </summary>
-    // NOTE: Duration is not supported by all systems,
-    // (i.e. iPhone) and cannot co-exist with DtEnd.
-    // RFC 5545 states:
-    //
-    //      ; either 'dtend' or 'duration' may appear in
-    //      ; a 'eventprop', but 'dtend' and 'duration'
-    //      ; MUST NOT occur in the same 'eventprop'
-    //
-    // Therefore, Duration is not serialized, as DtEnd
-    // should always be extrapolated from the duration.
-    public virtual TimeSpan Duration
+    /// Likewise, the duration is determined with start/end times
+    /// if it is not explicitly set.
+    /// <para/>
+    /// RFC 5545 states:
+    /// Either 'dtend' or 'duration' may appear in
+    /// an 'eventprop', but 'dtend' and 'duration'
+    /// MUST NOT occur in the same 'eventprop'
+    /// </remarks>
+    public virtual TimeSpan? Duration
     {
-        get => Properties.Get<TimeSpan>("DURATION");
+        get => Properties.Get<TimeSpan?>("DURATION");
         set
         {
-            if (!Equals(Duration, value))
-            {
-                Properties.Set("DURATION", value);
-            }
+            if (DtEnd is not null) throw new InvalidOperationException("DURATION property cannot be set when DTEND property is not null.");
+            SetProperty("DURATION", value); 
         }
     }
 
     /// <summary>
-    /// Calculates and returns the duration of the first occurrence of this event.
+    /// Calculates and returns the <b>nominal duration</b> of the first occurrence of this event.
+    /// <para/>
+    /// If the <see cref="CalendarEvent.Duration"/> property is not null, its value will be returned.<br/>
+    /// If <see cref="RecurringComponent.DtStart"/> and <see cref="CalendarEvent.DtEnd"/> are set, it will return <see cref="CalendarEvent.DtEnd"/> minus <see cref="CalendarEvent.DtStart"/>.<br/>
+    /// Otherwise, it will return <see cref="TimeSpan.Zero"/>.
     /// </summary>
     /// <remarks>
-    /// If the 'DURATION' property is set, this method will return its value.
-    /// Otherwise, if DTSTART and DTEND are set, it will return DTSTART minus DTEND.
-    /// Otherwise it will return `default(TimeSpan)`.
-    /// Note that for recurring events, the duration of individual occurrences may vary
-    /// if they span a DST change.
+    /// Note: For recurring events, the <b>effective duration</b> of individual occurrences may vary due to DST transitions
+    /// of the given <see cref="RecurringComponent.DtStart"/> and <see cref="CalendarEvent.DtEnd"/> timezones.
     /// </remarks>
-    /// <returns>The effective duration of this event.</returns>
-    public virtual TimeSpan GetFirstDuration()
+    /// <returns>The <b>nominal duration</b> of this event.</returns>
+    public virtual TimeSpan CalcFirstNominalDuration()
```

Comment:
Regarding the naming I'd rather go for just 'GetFirst...'. That calculations are done internally doesn't seem to be relevant from an outside perspective.

Reply:
If we improve the return value as suggested below, i.e. that it can represent the duration in a way that allows keeping nominal/exact apart, then we even wouldn't be talking about the 'first' duration anymore, so then a proper name could be like `GetEffectiveDuration()`. Like

`Duration GetEffectiveDuration()`

In `CalDateTime` we could then offer something like `Add(Duration d)` taking care of the difference between both parts.

Reply:
See the [comment](https://github.com/ical-org/ical.net/pull/673#issuecomment-2558053226)